backend/logic/users/users.js

```diff
@@ -14,33 +14,141 @@ var insertStudentUser = async (username, password, email, id) => {
     error = "invalid format password";
   } else if (!format.verifyEmail(email)) {
     error = "invalid format email";
-  } else if (!format.verifyId(id)) {
+  } else if (!format.verifyStudentId(id)) {
     error = "invalid format id";
   }
 
-  console.error(error)
   if (!error == false) {
     console.error(error);
-    //TODO replace this with an error when tests are fixed for it 
-    return error;
+    throw new Error(error);
+  }
+
+  let connection;
+  try {
+    connection = await mysql.getNewConnection();
+  } catch (error) {
+    throw new Error("failed to establish connection with database");
   }
 
   // Hash the password
   let hash = hasher.hashPass(password);
   try {
-    let connection = await mysql.getNewConnection()
     await connection.beginTransaction();
     await connection.query("INSERT INTO users VALUES(?, ?, ?);", [username, email, hash]);
     await connection.query("INSERT INTO students VALUES(?, ?);", [id, username]);
     await connection.commit();
     connection.release();
-    //TODO remove this after test refactoring 
     return true;
   } catch (error) {
     connection.rollback();
     connection.release();
     console.error(error);
-    return false;
+    throw new Error(false);
+  }
+};
+
+var deleteStudentUser = async (username) => {
+  if (!format.verifyUsername(username)) {
+    let error = "invalid username";
+    console.error(error);
+    throw new Error(error);
+  }
+
+  let connection;
+  try {
+    connection = await mysql.getNewConnection();
+  } catch (error) {
+    console.error(error);
+    throw new Error("failed to establish connection with database");
+  }
+
+  try {
+    await connection.beginTransaction();
+    let student_id = await connection.query('SELECT student_id FROM students WHERE username = ?;', username)
+    await connection.query('DELETE FROM students WHERE student_id = ?;', [student_id[0].student_id]);
+    await connection.query('DELETE FROM users WHERE username = ?;', [username]);
+    await connection.commit();
+    return true;
```

Comment:
Missing connection.release()?
```suggestion
    connection.release();
    return true;
```

Reply:
True good catch